### ISSUE TITLE
carthage update <dependency> with typo builds all dependencies

### ISSUE THREAD

**Issue by halseth** [Carthage/Carthage#1517]
* `0.18`: 
* `Xcode 7.3.1
Build version 7D1014`: 
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? Yes
* Are you using `--use-submodules`? No

**Cartfile**
```
github "<private-path>/AudioEngine" "release-v1.1"
github "<private-repo>" "v0.3.4"
github "yeahdongcn/UIColor-Hex-Swift" "1.8"
github "facebook/pop" "1.0.9"
github "ashleymills/Reachability.swift" "v2.3.2"
github "<private-repo>" "v0.2"
github "mixpanel/mixpanel-iphone" "v2.9.4"
github "auth0/JWTDecode.swift" "1.0.0"
```

The command `carthage update AudioEngine --no-use-binaries --platform iOS` does what I expect: It fetches and builds `AudioEngine` as listed in my Cartfile.

The command `carthage update AUdioEngine --no-use-binaries --platform iOS` however (note the uppercase typo), for some reason fetches and builds all dependencies in my Cartfile. I would expect it to throw an error saying the dependency is not found, or only build `AudioEngine`.

**Comment by hpwooten** [comment]
I've experienced the same behavior. I'm really enjoying Carthage, but I think one area of improvement would be to stop silently assuming other behavior than is intended, like failing on a typo.

**Comment by mdiep** [comment]
I agree. Care to open a pull request? 😄

**Comment by hpwooten** [comment]
I can take a look!

**Comment by netbe** [comment]
@hpwooten did you start something ? I'll be happy to help out too

**Comment by hpwooten** [comment]
@netbe I haven't started it yet. Got caught up in some other things.

**Comment by hpwooten** [comment]
@netbe Just followed you on Twitter. Let me know if you'd like to collaborate on a fix

**Comment by netbe** [comment]
@hpwooten yep taking a break during these holidays, but will let you know when I am ready to tackle this during january

**Comment by hpwooten** [comment]
Started work here: https://github.com/hpwooten/Carthage/tree/bugfix/unknown-dependency-error

Need to test a bit more to make sure it works as intended.

**Comment by bwhiteley** [comment]
We just got bit by this one again. We have a script that runs a series of carthage commands. One of the commands is specific to one framework. We removed that dependency from the project and forgot to update the script. Now the script runs the command meant for one framework on all of the dependencies. 

Desired behavior: Exit with an error if a specified dependency is not found.

**Comment by KelvinEC** [comment]
Any news about this issues?
